enterprise/server/ociregistry/ociregistry_test.go

```diff
@@ -0,0 +1,130 @@
+package ociregistry_test
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"net"
+	"net/http"
+	"runtime"
+	"testing"
+
+	"github.com/buildbuddy-io/buildbuddy/enterprise/server/ociregistry"
+	"github.com/buildbuddy-io/buildbuddy/enterprise/server/util/oci"
+	"github.com/buildbuddy-io/buildbuddy/server/environment"
+	"github.com/buildbuddy-io/buildbuddy/server/testutil/testenv"
+	"github.com/buildbuddy-io/buildbuddy/server/testutil/testport"
+	"github.com/buildbuddy-io/buildbuddy/server/testutil/testregistry"
+	"github.com/buildbuddy-io/buildbuddy/server/util/testing/flags"
+	v1 "github.com/google/go-containerregistry/pkg/v1"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	rgpb "github.com/buildbuddy-io/buildbuddy/proto/registry"
+)
+
+func runTestProxy(t *testing.T, env environment.Env) string {
+	t.Helper()
+	r, err := ociregistry.New(env)
+	require.Nil(t, err)
+	port := testport.FindFree(t)
+
+	mux := http.NewServeMux()
+	mux.Handle("/", r)
+
+	listenAddr := fmt.Sprintf("localhost:%d", port)
+	server := &http.Server{Handler: r}
+	lis, err := net.Listen("tcp", listenAddr)
+	require.NoError(t, err)
+	go func() { _ = server.Serve(lis) }()
+	t.Cleanup(func() {
+		server.Shutdown(context.TODO())
+	})
+	return listenAddr
+}
+
+func TestResolve(t *testing.T) {
+	te := testenv.GetTestEnv(t)
+
+	registry := testregistry.Run(t, testregistry.Opts{})
+	imageName, randomImage := registry.PushRandomImage(t)
+	proxyAddr := runTestProxy(t, te)
+
+	flags.Set(t, "executor.container_registry_mirrors", []oci.MirrorConfig{{
+		OriginalURL: "http://" + registry.Address(),
+		MirrorURL:   "http://" + proxyAddr,
+	}})
+
+	resolvedImage, err := oci.Resolve(
+		context.Background(),
+		imageName,
+		&rgpb.Platform{
+			Arch: runtime.GOARCH,
+			Os:   runtime.GOOS,
+		},
+		oci.Credentials{})
+	require.NoError(t, err)
+	assertSameImages(t, randomImage, resolvedImage)
```

Comment:
is there a way to assert that the image was downloaded via the proxy? at a quick glance, it looks like this test would pass without setting the `executor.container_registry_mirrors` flag

Reply:
Good point! The test does indeed pass if you comment out the flag.
I think I'll have to write a separate test that uses the `go-containerregistry` library directly to explicitly pull from the mirror. The `Image` itself does not contain information about where it was fetched from (which makes sense, it's an immutable container image).

Reply:
in `runTestProxy`, we could also wrap the handler `r` with a small handler that increments a counter or something like that, then the test could assert that the proxy handled at least one HTTP request. That way we're testing the real behavior of pulling using the normal image ref (rather than explicitly pulling from the mirror), while being somewhat confident that the proxy was involved in handling the request

Reply:
Added some gross code to the test to make sure mirror receives some requests and the test registry receives at least that many. One request to the mirror generates multiple requests to the test registry.